core/ledger/kvledger/txmgmt/validation/batch_preparer.go

```diff
@@ -19,20 +19,95 @@ import (
 	"github.com/hyperledger/fabric/core/ledger/kvledger/txmgmt/rwsetutil"
 	"github.com/hyperledger/fabric/core/ledger/kvledger/txmgmt/statedb"
 	"github.com/hyperledger/fabric/core/ledger/kvledger/txmgmt/txmgr"
-	"github.com/hyperledger/fabric/core/ledger/kvledger/txmgmt/validator"
-	"github.com/hyperledger/fabric/core/ledger/kvledger/txmgmt/validator/internal"
 	"github.com/hyperledger/fabric/core/ledger/util"
 	"github.com/hyperledger/fabric/internal/pkg/txflags"
 	"github.com/hyperledger/fabric/protoutil"
 )
 
+// CommitBatchPreparer performs validation and prepares the final batch that is to be committed to the statedb
+type CommitBatchPreparer struct {
+	txmgr              txmgr.TxMgr
+	db                 privacyenabledstate.DB
+	validator          *validator
+	customTxProcessors map[common.HeaderType]ledger.CustomTxProcessor
+}
+
+// NewCommitBatchPreparer constructs a validator that internally manages statebased validator and in addition
+// handles the tasks that are agnostic to a particular validation scheme such as parsing the block and handling the pvt data
+func NewCommitBatchPreparer(
+	txmgr txmgr.TxMgr,
+	db privacyenabledstate.DB,
+	customTxProcessors map[common.HeaderType]ledger.CustomTxProcessor,
+	hasher ledger.Hasher,
+) *CommitBatchPreparer {
+	return &CommitBatchPreparer{
+		txmgr,
+		db,
+		&validator{
+			DB:     db,
+			Hasher: hasher,
+		},
+		customTxProcessors,
+	}
+}
+
+// ValidateAndPrepareBatch implements the function in interface validator.Validator
+func (impl *CommitBatchPreparer) ValidateAndPrepareBatch(blockAndPvtdata *ledger.BlockAndPvtData,
+	doMVCCValidation bool) (*privacyenabledstate.UpdateBatch, []*txmgr.TxStatInfo, error) {
+	block := blockAndPvtdata.Block
+	logger.Debugf("ValidateAndPrepareBatch() for block number = [%d]", block.Header.Number)
+	var internalBlock *Block
+	var txsStatInfo []*txmgr.TxStatInfo
+	var pubAndHashUpdates *PubAndHashUpdates
```

Comment:
Though `PubAndHashUpdates` can be unexported, we can export it as it encapsulates exported fields. If you feel the same, we can retain it as-is.